src/generate.ts

```diff
@@ -97,21 +101,40 @@ const recurseRelationships = (
 			const [localField, referencedField] = (
 				relationships as RelationshipLeafNode
 			).$;
-			if (currentTable.idField === localField && referencedField != null) {
-				const referencedTable = m.tables[referencedField[0]];
-				if (referencedTable != null) {
-					const referencedInterface = getReferencedInterface(
-						referencedTable.name,
-						mode,
-					);
-					const propDefinitons = [
-						`${parentKey}?: Array<${referencedInterface}>;`,
-					];
-					const synonym = inverseSynonyms[odataNameToSqlName(parentKey)];
-					if (synonym != null) {
-						propDefinitons.push(
-							`${sqlNameToODataName(synonym)}?: Array<${referencedInterface}>;`,
+			if (referencedField != null) {
```

Comment:
What do you think about instead doing `if (referencedField == null) return [];` to avoid deep nesting

Reply:
sure